persistence-cassandra/core/src/main/scala/com/lightbend/lagom/internal/persistence/cassandra/CassandraKeyspaceConfig.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.persistence.cassandra
+
+import akka.event.LoggingAdapter
+import com.typesafe.config.Config
+
+private[lagom] object CassandraKeyspaceConfig {
+
+  def validateKeyspace(namespace: String)(implicit config: Config, log: LoggingAdapter): Unit =
+    validateKeyspace(namespace, s"$namespace.defaults")
+
+  def validateKeyspace(namespace: String, defaultNamespace: String)(implicit config: Config, log: LoggingAdapter): Unit = {
+    val path = s"$namespace.keyspace"
+    if (!config.hasPath(path)) {
+      log.error("Configuration for [{}] must be set in application.conf", path)
+    } else if (log.isWarningEnabled) {
+      val keyspace = config.getString(path)
+      val defaultKeyspace = config.getString(s"$defaultNamespace.keyspace")
+      if (keyspace == defaultKeyspace) {
```

Comment:
So the problem is that we're going to remove the defaults, which collapses 1 & 2 into the same case.

The cases we're really interested in are:

1. The default value is inherited implicitly
2. The value is provided explicitly, but happens to be the same as the default value
3. The value is provided explicitly, and is different than the default value

We want to warn on 1 because it's relying on an implicitly-selected default. If the default changes or goes away, the keyspace used will change, which could be fairly disastrous with default settings running in production. That's the motivation for the approach I'm taking to this change.

2 & 3 are both valid, but there's no way to tell the difference between 1 & 2

Whether or not the default is the same as the reference value is less important in my mind. Sharing keyspaces between services is bad for sure, but relying on an implicit keyspace is bad whether or not they're shared. You want the user to be in control of where their data goes. If they choose a shared keyspace, that's on them. AFAICT, it doesn't cause any actual conflicts if you don't use the same entity names, read side tables, etc.